nixos/modules/services/networking/firewall.nix

```diff
@@ -493,6 +508,22 @@ in
         '';
     };
 
+    networking.firewall.extraAllowed = mkOption {
+      type = types.listOf (types.submodule { options = {
+           source = mkOption { type = types.string; default = ""; example = "example.com"; description = "Source of the packets"; };
+           protocol = mkOption { type = types.enum ["tcp" "udp" "icmp"]; default = ""; example = "tcp"; description = "Protocol used."; };
+           dport = mkOption { type = types.int; default = 0; example = 22; description = "Destination port of the packet."; };
```

Comment:
I find `destinationPort` better here. The nixos module should be easier to use then iptables.
`sourcePort` might be also worth adding.